Zend/zend_inheritance.c

```diff
@@ -200,22 +200,39 @@ static bool class_visible(zend_class_entry *ce) {
 	}
 }
 
+static zend_always_inline void register_unresolved_class(zend_string *name) {
+	/* We'll autoload this class and process delayed variance obligations later. */
+	if (!CG(delayed_autoloads)) {
+		ALLOC_HASHTABLE(CG(delayed_autoloads));
+		zend_hash_init(CG(delayed_autoloads), 0, NULL, NULL, 0);
+	}
+	zend_hash_add_empty_element(CG(delayed_autoloads), name);
+}
+
 static zend_class_entry *lookup_class(
 		zend_class_entry *scope, zend_string *name, bool register_unresolved) {
-	uint32_t flags = ZEND_FETCH_CLASS_ALLOW_UNLINKED | ZEND_FETCH_CLASS_NO_AUTOLOAD;
-	zend_class_entry *ce = zend_lookup_class_ex(name, NULL, flags);
+	if (UNEXPECTED(!EG(active))) {
+		if (zend_string_equals_ci(scope->name, name)) {
+			return scope;
+		}
+
+		if (register_unresolved) {
+			register_unresolved_class(name);
+		}
+
+		return NULL;
+	}
```

Comment:
Not sure this is really the right way to go about this. The way I see it, the core problem is that zend_lookup_class_ex is using EG(class_table), while at this point in time only CG(class_table) is initialized.

TBH I'm not sure why we even have two separate globals for this.

Reply:
zend_lookup_class_ex referencing EG(class_table) means you can't execute it before the executor is active, would appear to be a reasonable description of what's going on also, no ?